### PR TITLE
Request parameter iterator

### PR DESCRIPTION
**Pull request by incubos**

### CODE REVIEW

File: src/one/nio/http/Request.java [hunk 1]
```diff
@@ -169,6 +170,94 @@ public String getRequiredParameter(String key) {
...
+                    return cur;
+                }
+            }
+
+            @Override
+            public void remove() {
+                throw new UnsupportedOperationException();
+            }
+        };
+    }
```
Comment (apangin) [inline src/one/nio/http/Request.java]: You may optionally add a method that returns `Iterable`, so that the iterator can be used in a for-each loop.

Comment (incubos) [reply src/one/nio/http/Request.java]: Done.

File: src/one/nio/http/QueryParameterIterator.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package one.nio.http;
```
Comment (apangin) [inline src/one/nio/http/QueryParameterIterator.java]: New source files should include standard license header

File: src/one/nio/http/QueryParameterIterator.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        }
+
+        assert !parameter.isEmpty();
+
+        return parse(parameter);
+    }
+
+    /**
+     * Skip '&'s and position at the next parameter or stops
+     */
```
Comment (apangin) [inline src/one/nio/http/QueryParameterIterator.java]: This is not a javadoc. Please convert to a regular comment.
The same applies to all other comments to private members.

Comment (incubos) [reply src/one/nio/http/QueryParameterIterator.java]: Fixed.

File: src/one/nio/http/QueryParameterIterator.java (hunk 2 above)
Comment (incubos) [reply src/one/nio/http/QueryParameterIterator.java]: Fixed.

**Pull request merged**